python/ray/dashboard/client/src/pages/dashboard/memory/Memory.tsx

```diff
@@ -0,0 +1,110 @@
+import {
+  createStyles,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  Theme,
+  withStyles,
+  WithStyles,
+  Button,
+} from "@material-ui/core";
+import React from "react";
+import { StoreState } from "../../../store";
+import { connect } from "react-redux";
+import MemoryRowGroup from "./MemoryRowGroup";
+import { dashboardActions } from "../state";
+
+const styles = (theme: Theme) =>
+  createStyles({
+    table: {
+      marginTop: theme.spacing(1),
+    },
+    cell: {
+      padding: theme.spacing(1),
+      textAlign: "center",
+      "&:last-child": {
+        paddingRight: theme.spacing(1),
+      },
+    },
+  });
+
+const mapStateToProps = (state: StoreState) => ({
+  tab: state.dashboard.tab,
+  memoryTable: state.dashboard.memoryTable,
+});
+
+const mapDispatchToProps = dashboardActions;
+
+type State = {
+  // If memory table is captured, it should stop reneweing memory table.
+  memoryTableCapture: boolean;
+};
+
+class MemoryInfo extends React.Component<
+  WithStyles<typeof styles> &
+    ReturnType<typeof mapStateToProps> &
+    typeof mapDispatchToProps,
+  State
+> {
+  state: State = {
+    memoryTableCapture: false,
+  };
+
+  handleMemoryTableCapture = () => {
+    this.setState((state) => ({
+      memoryTableCapture: !state.memoryTableCapture,
+    }));
+    this.props.setShouldObtainMemoryTable(this.state.memoryTableCapture);
+  };
+
+  render() {
+    const { classes, memoryTable } = this.props;
+    // console.log(memoryTable);
+    return (
+      <React.Fragment>
+        {memoryTable !== null ? (
+          <React.Fragment>
+            <Button color="primary" onClick={this.handleMemoryTableCapture}>
+              {this.state.memoryTableCapture
+                ? "Stop capturing"
+                : "Capture Memory Table"}
+            </Button>
+            <Table className={classes.table}>
+              <TableHead>
+                <TableRow>
+                  <TableCell className={classes.cell} />
+                  <TableCell className={classes.cell}>IP Adress</TableCell>
+                  <TableCell className={classes.cell}>Pid</TableCell>
+                  <TableCell className={classes.cell}>Type</TableCell>
+                  <TableCell className={classes.cell}>Object ID</TableCell>
+                  <TableCell className={classes.cell}>Object Size</TableCell>
+                  <TableCell className={classes.cell}>Reference Type</TableCell>
+                  <TableCell className={classes.cell}>Call Site</TableCell>
```

Comment:
If the table cells won't need to be rendered differently for each item, we can reduce code duplication by doing:
```
['IP Address', 'Pid', ...].map((header) => <TableCell className={classes.cell}>{header}</TableCell>
```

Regardless, make sure to fix the spelling of 'IP Address'